rundeckapp/grails-spa/packages/ui-trellis/src/library/components/widgets/rundeck-info/RundeckInfo.spec.ts

```diff
@@ -0,0 +1,110 @@
+    expect(wrapper.findComponent({ name: "RundeckLogo" }).exists()).toBe(true);
+    expect(wrapper.findComponent({ name: "PagerdutyLogo" }).exists()).toBe(
+      false,
+    );
+  });
+  it("renders PagerdutyLogo when appInfo.title is 'Pagerduty'", async () => {
+    const wrapper = await mountRundeckInfo({ appInfo: { title: "Pagerduty" } });
+
+    expect(wrapper.findComponent({ name: "RundeckLogo" }).exists()).toBe(false);
+    expect(wrapper.findComponent({ name: "PagerdutyLogo" }).exists()).toBe(
+      true,
+    );
+  });
+  it("renders RundeckVersions information correct props based on version data", async () => {
+    const versionProps = {
+      logo: false,
+      logocss: "some-css",
+      number: "1.0.0",
+      tag: "stable",
+      title: "Rundeck",
+    };
+    const wrapper = await mountRundeckInfo({ version: versionProps });
+
+    await wrapper.vm.$nextTick();
+    const versionComponent = wrapper.findComponent({ name: "RundeckVersion" });
+    expect(versionComponent.exists()).toBe(true);
+    expect(versionComponent.props()).toEqual(versionProps);
+  });
+  it("renders ServerDisplay correctly based on a server props", async () => {
+    const serverProps = {
+      name: "localHost",
+      icon: "paperclip",
+      uuid: "uuid1",
+      showId: true,
+    };
+    const wrapper = await mountRundeckInfo({ server: serverProps });
+    await wrapper.vm.$nextTick();
+    const serverComponent = wrapper.findComponent({ name: "ServerDisplay" });
+    expect(serverComponent.exists()).toBe(true);
+    expect(serverComponent.props()).toEqual({
+      name: serverProps.name,
+      glyphicon: serverProps.icon,
+      uuid: serverProps.uuid,
+      showId: serverProps.showId,
+    });
+  });
+
+  it("renders version display correctly", async () => {
+    const versionDisplay = {
+      name: "Rundeck",
+      icon: "icon-url",
+      color: "lamp",
+    };
+    const wrapper = await mountRundeckInfo({ version: versionDisplay });
+    const versionDisplayComponent = wrapper.findComponent({
+      name: "VersionDisplay",
+    });
+
+    expect(versionDisplayComponent.props("text")).toBe("Rundeck lamp icon-url");
+  });
+
+  it("renders latest release information correctly ", async () => {
+    const latestProps = {
+      full: "2.0.0",
+      tag: "GA",
+    };
+    const wrapper = await mountRundeckInfo({
+      latest: latestProps,
+    });
+
+    const latestSection = wrapper
+      .find(".rundeck-info-widget__latest")
+      .findComponent({ name: "RundeckVersion" });
+    expect(latestSection.exists()).toBe(true);
+
+    expect(latestSection.props("number")).toEqual(latestProps.full);
+    expect(latestSection.props("tag")).toEqual(latestProps.tag);
+  });
+
+  it("should have an anchor with the correct href", async () => {
+    const wrapper = await mountRundeckInfo();
+    const expectedHref = "http://localhost";
+    const anchorElement = wrapper.find(`a[href="${expectedHref}"]`);
```

Comment:
So ideally here you don't want to select the a by the property you're about to test, but by a data-test-id (it's more resilient. Changes to do:

- add a data-test-id for the `<a>` you're going to test in `RundeckInfo.vue`;
- update test to find this a by the data-test-id;

Reply:
> So ideally here you don't want to select the a by the property you're about to test, but by a data-test-id (it's more resilient. Changes to do:
> 
> * add a data-test-id for the `<a>` you're going to test in `RundeckInfo.vue`;
> * update test to find this a by the data-test-id;

Fixed